### ISSUE TITLE
Workspace switches when a window is closed and same app is active on another workspace

### ISSUE THREAD

**Issue by sschuldenzucker** [nikitabobko/AeroSpace#854]
**Describe the bug**

Description: Consider a single-monitor setup. Let `A` be an app and `a` and `b` be workspaces. If workspace `a` is focused and has *only* windows of app `A` on it and the before-that-most-recently-focused window of app `A` is on `b`, and the current window (of app `A`) on `a` is closed, then the workspace is switched to `b`. This does not happen if `a` has a window from a different app on it.

Steps to reproduce:
- Pick any app (say, Safari)
- Open a Safari window on workspace 1
- Open two Safari windows on workspace 2
- Switch to workspace 1
- Switch to workspace 2
- Close the current window (on workspace 2)

Expected behavior: Workspace 2 remains active and the remaining Safari window is focused.

Actual behavior: Workspace 1 becomes active and its Safari window is focused.

This also happens when workspace 2 only has one window, i.e., with the following steps:
- Pick any app (say, Safari)
- Open a Safari window on workspace 1
- Open one Safari windows on workspace 2
- Switch to workspace 1
- Switch to workspace 2
- Close the current window (on workspace 2)

Expected behavior: Workspace 2 remains active (empty).

Actual behavior: Workspace 1 becomes active and its Safari window is focused.

**Related issues you could find**

This is a variant of #571 and probably more-or-less the same issue.

**Aerospace version**

```
aerospace CLI client version: 0.16.2-Beta 21336ad382539b35fdc94b4fbd55408e10b101f8
AeroSpace.app server version: 0.16.2-Beta 21336ad382539b35fdc94b4fbd55408e10b101f8
```

**Comment by mkhattat** [comment]
I believe this is due to the same race condition mentioned in #571. In that case, couldn't we simply ignore focus events from MacOS when closing the last app in the workspace?

**Comment by Haerezis** [comment]
I agree that this is a bit annoying, and I would love if there was a fix to that.

**Comment by nikitabobko** [comment]
Yes, it's known and unfortunate problem. I've described it here: https://github.com/nikitabobko/AeroSpace/issues/1097

Closing the issue in favor of https://github.com/nikitabobko/AeroSpace/issues/1097
